mdspan_copy/mdspan_copy.md

```diff
@@ -0,0 +1,78 @@
+---
+title: "Copy and fill for `mdspan`"
+date: today
+---
+
+# Motivation
+
+C++23 introduced `mdspan` ([@P0009R18]), a nonowning multidmensional array abstraction that has a customizable layout. Layout customization was originally motivated in [@P0009R18] with considerations for interoperability and performance, particularly on different architectures. Moreover, [@P2630R4] introduced `submdspan`, a slicing function that can yield arbitrarily strided layouts. Without standard library support, copying efficiently between mdspans with mixes of complex layouts is challenging for users.
+
+Many applications, including high-performance computing (HPC), image processing, computer graphics, etc that benefit from `mdspan` also would benefit from basic memory operations provided in standard algorithms such as copy and fill. Indeed, the authors found that a copy algorithm would have been quite useful in their implementation of the copying `mdarray` ([@P1684]) constructor.
+
+However, existing standard library facilities are not sufficient here. Currently, `mdspan` does not have iterators or ranges that represent the span of the `mdspan`. Additionally, it's not entirely clear what this would entail.
+
+Moreover, the manner in which an `mdspan` is copied (or filled) is highly performance sensitive, particularly in regards to caching behavior when traversing mdspan memory. A naïve user implementation is easy to get wrong in addition to being tedious for higher rank `mdspan`s. Ideally, an implementation would be free to use information about the layout of the `mdspan` known at compile time to perform optimizations; e.g. a continuous span `mdspan` copy for trivial types could be implementeed with a `memcpy`.
```

Comment:
Naive implementation is not tedious
```C++
for i0 in [0, extents(0)
  for i1 in [0, extents(1)
    ...
      for iK in [0, extents(K)
  dst[i0, i1, ..., iK] = src[i0, i1, ..., iK];
```
just really inefficient.

Reply:
Haha well, I think writing an 8-nested for loop is pretty tedious :P. I guess it's a matter of opinion